### PR TITLE
Add DartDevcBootstrapTransformer and compiler tests for dartdevc

### PR DESCRIPTION
**Pull request by jakemac53**
This is the final PR for the basic functionality. It also includes a few bug fixes for the previous transformers based on additional tests this adds, as well as issues discovered when running pop_pop_win and angular_components.

After this pop_pop_win is fully functional (tests work too via pub serve). The angular components example app has some issues relating to the angular transformer which I am currently investigating (it fundamentally does work though).

### CODE REVIEW

File: lib/src/barback/dartdevc/linked_summary_transformer.dart [hunk 1]
```diff
@@ -69,10 +69,12 @@ Future _createLinkedSummaryForModule(
       p.url.join(outputDir, '${module.id.name}$linkedSummaryExtension'));
   var summaryOutputFile = tempEnv.fileFor(summaryOutputId);
   var request = new WorkRequest();
+  // TODO(jakemac53): Diet parsing results in erroneous errors later on today,
```
Comment (munificent) [inline lib/src/barback/dartdevc/linked_summary_transformer.dart]: "Later on today"?

File: test/compiler/compiler_flag_test.dart [hunk 2]
```diff
@@ -29,20 +30,36 @@ main() {
...
+        requestShouldSucceed('web__main.js.map', contains('web__main.js'));
+        requestShouldSucceed('dart_sdk.js', null);
+        requestShouldSucceed('require.js', null);
+        requestShouldSucceed('main.dart.js', null);
+        break;
+      case Compiler.dart2JS:
+        requestShouldSucceed('main.dart.js', null);
+        requestShould404('web__main.js');
+        // requestShould404('dart_sdk.js');
+        // requestShould404('require.js');
```
Comment (munificent) [inline test/compiler/compiler_flag_test.dart]: Delete these lines.

File: lib/src/barback/dartdevc/linked_summary_transformer.dart (hunk 1 above)
Comment (jakemac53) [inline lib/src/barback/dartdevc/linked_summary_transformer.dart]: sorry weird wording lol, reworded it

File: test/compiler/compiler_flag_test.dart (hunk 2 above)
Comment (jakemac53) [inline test/compiler/compiler_flag_test.dart]: done

**Pull request merged**